idea/src/org/jetbrains/kotlin/idea/quickfix/ChangeFunctionSignatureFix.kt

```diff
@@ -59,11 +60,12 @@ abstract class ChangeFunctionSignatureFix(
     }
 
     protected fun getNewArgumentName(argument: ValueArgument, validator: Function1<String, Boolean>): String {
-        val argumentName = argument.getArgumentName()
+        val argumentName = argument.getArgumentName()?.asName?.asString()
+            ?: (argument.getArgumentExpression() as? KtNameReferenceExpression)?.getReferencedName()?.takeIf { it != "it" }
```

Comment:
I suppose, `field` is the another special case:

```
fun foo() {}

class Test {
    val x: String = ""
        get() {
            foo(field)
            return field
        }
}
```

Maybe we can even take the property name instead.

Reply:
Maybe also deparenthesize the name expression?